src/main/java/io/vertx/core/buffer/impl/VertxByteBufAllocator.java

```diff
@@ -11,22 +11,52 @@
 package io.vertx.core.buffer.impl;
 
 import io.netty.buffer.AbstractByteBufAllocator;
+import io.netty.buffer.AdaptiveByteBufAllocator;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.PooledByteBufAllocator;
 import io.netty.buffer.UnpooledByteBufAllocator;
 import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.SystemPropertyUtil;
 
 public abstract class VertxByteBufAllocator extends AbstractByteBufAllocator {
 
   /**
    * Vert.x pooled allocator.
    */
-  public static final ByteBufAllocator POOLED_ALLOCATOR = new PooledByteBufAllocator(true);
+  public static final ByteBufAllocator POOLED_ALLOCATOR = pooledAllocator();
+
+  /**
+   * Tries hard to reuse the existing default Netty allocators instances to avoid creating new instances.
+   */
+  private static ByteBufAllocator pooledAllocator() {
+    String allocType = allocType();
```

Comment:
I think we should log which one gets finally used

Reply:
That's a good idea...trace?

Reply:
I think debug is OK, it appears only once in the lifetime of the JVM

Reply:
FIY: netty is already logging this, is it really worth it to double that?

Reply:
Good point: found https://github.com/netty/netty/blob/netty-4.1.112.Final/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L80-L98